src/extension/inpage-script/index.js

```diff
@@ -1,8 +1,10 @@
 import WebLNProvider from "../ln/webln";
+import LNURLProvider from "../ln/lnurl";
 import Donation from "./donation";
 
 if (document) {
   window.webln = new WebLNProvider();
+  window.lnurl = new LNURLProvider();
```

Comment:
actually I think we should stick with WebLN here and propose a new `lnul` function on the webln object. 

so if it is a `lnurl` link we call the `webln.lnurl()` function which sends the message to lnurl action. There we decode the lnurl, do the GET request and then continue depending on the `type` of the lnurl (pay, auth, etc.)

what do you think?

Reply:
The lnurl action sounds good to me. (Moving the decoding and GET away from React) I'm only not sure about if lnurl should be part of webln as they are different directions?

Reply:
yes, that needs to be discussed. Though I am not sure if it is different directions. WebLN describes a way how to interact with lightning from the web/JS. 
Instead of passing a lightning invoice we are passing in a lnurl. 

(somewhat related discussion around extending webln with keysend: https://github.com/joule-labs/webln/issues/33)